### PR TITLE
[Examples] AWS Neuron Accelerator Example.

### PR DESCRIPTION
**Pull request by cblmemo**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: examples/neuron/inferentia.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+resources:
+  cloud: aws
+  accelerators: Inferentia:6
+  disk_size: 512
+  ports: 9000
+
+envs:
+  MODEL_NAME: meta-llama/Meta-Llama-3-8B-Instruct
+  HF_TOKEN: # fill
+  NEURON_RT_VISIBLE_CORES: "0-5"
```
Comment (Michaelvll) [inline examples/neuron/inferentia.yaml:10]: If possible, we should be able to automatically set this, as we did for `CUDA_VISIBLE_DEVICE` when people specify different number of GPUs to use for a task.

File: examples/neuron/inferentia.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+resources:
```
Comment (Michaelvll) [inline examples/neuron/inferentia.yaml:1]: Can we create another yaml that allows multiple accelerators, e.g,`accelerators: {A100:4, Inferentia:6}`? This way people can allow the system to choose whatever accelerators available.

The YAML can be brute force, i.e., do a `if..else` in setup and run.

File: sky/clouds/aws.py [hunk 3]
```diff
@@ -225,6 +225,9 @@ def _get_default_ami(cls, region_name: str, instance_type: str) -> str:
             if acc_name == 'K80':
                 image_id = service_catalog.get_image_id_from_tag(
                     'skypilot:k80-ubuntu-2004', region_name, clouds='aws')
+            if acc_name in ['Trainium', 'Inferentia']:
+                image_id = service_catalog.get_image_id_from_tag(
+                    'skypilot:neuron-ubuntu-2004', region_name, clouds='aws')
```
Comment (Michaelvll) [inline sky/clouds/aws.py:230]: Maybe just let's use `2204`, considering we are updating the system version in #3887

**Review by Michaelvll — COMMENTED** [review]
Thanks for adding this example @cblmemo! Left a few comments.

**Comment by cblmemo** [comment]
@Michaelvll Resolved all comments. PTAL again!

File: examples/neuron/inferentia.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
```
Comment (Michaelvll) [inline examples/neuron/inferentia.yaml:1]: Maybe call this folder `aws-neuron`?

**Review by Michaelvll — APPROVED** [review]
Thanks for adding this example @cblmemo! LGTM

**Pull request merged**